vcf.c

```diff
@@ -451,7 +452,8 @@ int bcf_hdr_register_hrec(bcf_hdr_t *hdr, bcf_hrec_t *hrec)
 
     // INFO/FILTER/FORMAT
     char *id = NULL;
-    int type = -1, num = -1, var = -1, idx = -1;
+    uint32_t type = -1, var = -1;
```

Comment:
Can I suggest we use UINT32_MAX from stdint.h instead? Things like this make me shiver.